update-ropsten-subgraph.js

```diff
@@ -0,0 +1,123 @@
+const Web3 = require("web3")
+const HDWalletProvider = require("@truffle/hdwallet-provider")
+const nunjucks = require('nunjucks');
+const fs = require('fs');
+
+const { readFileSync } = require("fs")
+const { resolve } = require("path")
+
+const networkID = 3 // Ropsten
+const keepCorePackageName = "@keep-network/keep-core"
+const keepEcdsaPackageName = "@keep-network/keep-ecdsa"
+const keepTbtcPackageName = "@keep-network/tbtc"
+
+const web3 = new Web3(process.env.ETH_RPC_URL)
+
+let ropstenVars
+
+// This is for ropsten network only
+async function readDeployedArtifacts() {
+
+  async function getContractData(packageName, contractName) {
+    let artifactRaw
+    try {
+      artifactRaw = readFileSync(
+        resolve(`./node_modules/${packageName}/artifacts/${contractName}.json`)
+      )
+    } catch (err) {
+      throw new Error(`failed to read artifact file: ${err.message}`)
+    }
+
+    let artifact
+    try {
+      artifact = JSON.parse(artifactRaw)
+    } catch (err) {
+      throw new Error(`failed to parse artifact file: ${err.message}`)
+    }
+
+    if (!artifact.networks[networkID]) {
+      throw new Error(
+        `configuration for network ${networkID} not found in ${contractName}`
+      )
+    }
+
+    if (!artifact.networks[networkID].address) {
+      throw new Error(
+        `missing address for network ${networkID} in ${contractName}`
+      )
+    }
+
+    if (!artifact.networks[networkID].transactionHash) {
+      throw new Error(
+        `missing transactionHash for network ${networkID} in ${contractName}`
+      )
+    }
+
+    const deployTransaction = await web3.eth.getTransaction(
+      artifact.networks[networkID].transactionHash
+    )
```

Comment:
Maybe it would be worth wrapping it with try/catch as well?

Reply:
Done in 010084b